src/code/z_player_lib.c

```diff
@@ -506,142 +558,684 @@ s32 func_8012364C(PlayState* play, Player* player, s32 arg2) {
-               : (gSaveContext.hudVisibility == HUD_VISIBILITY_A_B_C)
-                   ? gSaveContext.save.equips.buttonItems[0][EQUIP_SLOT_C_DOWN]
-                   : ITEM_NONE;
+        return (gSaveContext.buttonStatus[EQUIP_SLOT_C_DOWN] != BTN_DISABLED) ? BUTTON_ITEM_EQUIP(0, EQUIP_SLOT_C_DOWN)
+               : (gSaveContext.hudVisibility == HUD_VISIBILITY_A_B_C)         ? BUTTON_ITEM_EQUIP(0, EQUIP_SLOT_C_DOWN)
+                                                                              : ITEM_NONE;
     }
 
-    return (gSaveContext.buttonStatus[3] != BTN_DISABLED) ? gSaveContext.save.equips.buttonItems[0][EQUIP_SLOT_C_RIGHT]
-           : (gSaveContext.hudVisibility == HUD_VISIBILITY_A_B_C)
-               ? gSaveContext.save.equips.buttonItems[0][EQUIP_SLOT_C_RIGHT]
-               : ITEM_NONE;
+    return (gSaveContext.buttonStatus[EQUIP_SLOT_C_RIGHT] != BTN_DISABLED) ? BUTTON_ITEM_EQUIP(0, EQUIP_SLOT_C_RIGHT)
+           : (gSaveContext.hudVisibility == HUD_VISIBILITY_A_B_C)          ? BUTTON_ITEM_EQUIP(0, EQUIP_SLOT_C_RIGHT)
+                                                                           : ITEM_NONE;
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80123810.s")
+u16 sCItemButtons[] = { BTN_CLEFT, BTN_CDOWN, BTN_CRIGHT };
+
+s32 func_80123810(PlayState* play) {
+    Player* player = GET_PLAYER(play);
+    s32 temp_v0;
+    ItemId itemId;
+    s32 i;
+
+    if (gSaveContext.save.unk_06 == 0) {
+        if (CHECK_BTN_ANY(CONTROLLER1(&play->state)->press.button, BTN_A | BTN_B)) {
+            play->interfaceCtx.unk_222 = 0;
+            play->interfaceCtx.unk_224 = 0;
+            Interface_SetHudVisibility(play->msgCtx.unk_120BC);
+            return -1;
+        }
+    } else {
+        gSaveContext.save.unk_06--;
+    }
 
-extern u8 sActionModelGroups[];
+    for (i = 0; i < ARRAY_COUNT(sCItemButtons); i++) {
+        if (CHECK_BTN_ALL(CONTROLLER1(&play->state)->press.button, sCItemButtons[i])) {
+            i++;
+            itemId = func_8012364C(play, player, i);
 
-s32 Player_ActionToModelGroup(Player* player, s32 actionParam) {
-    s32 modelGroup = sActionModelGroups[actionParam];
+            play->interfaceCtx.unk_222 = 0;
+            play->interfaceCtx.unk_224 = 0;
+            Interface_SetHudVisibility(play->msgCtx.unk_120BC);
 
-    if ((modelGroup == 2) && Player_IsGoronOrDeku(player)) {
-        return 1;
+            if ((itemId >= ITEM_FD) || (temp_v0 = play->unk_18794(play, player, itemId, i), (temp_v0 < 0))) {
+                play_sound(NA_SE_SY_ERROR);
+                return -1;
+            } else {
+                s32 pad;
+
+                player->heldItemButton = i;
+                return temp_v0;
+            }
+        }
+    }
+
+    return 0;
+}
+
+// Used to map action params to model groups
+u8 sActionModelGroups[PLAYER_AP_MAX] = {
+    PLAYER_MODELGROUP_DEFAULT,        // PLAYER_AP_NONE
```

Comment:
Where do the names in the comments come from?

Reply:
If you mean the `PLAYER_AP_NONE` and others they are the `PlayerActionParam` enum, which is used as an index of this array